manta-pay/src/crypto/constraint/arkworks/pairing.rs

```diff
@@ -175,3 +175,163 @@ pub mod bls12 {
         type Deserialize = G2Prepared<P>;
     }
 }
+
+/// Bn254 Utilities
```

Comment:
How different is this from the BLS one? If it's the same, please write a macro that generates this code for an arbitrary model and then instantiate it for each one.